File: Code/Python/problem002/euler002.py
#!/usr/bin/env python3


class Fibo():
    def __init__(self):
        self.fiboList = [1]

    def addFiboNumber(self):
        l = len(self.fiboList)
        if 0 == l:
            self.fiboList = [1]
        elif 1 == l:
            self.fiboList = [1, 1]
        else:
            self.fiboList.append(self.fiboList[l - 1] + self.fiboList[l - 2])

    def getFiboNumber(self, index):
        return self.fiboList[index]

    def getFiboList(self):
        return self.fiboList


fibo = Fibo()

index = 0

max = 4000000

val = fibo.getFiboNumber(index)

while (val < max):
    fibo.addFiboNumber()
    index = index + 1
    val = fibo.getFiboNumber(index)

fibos = fibo.getFiboList()

sum = 0

for i in fibos:
    if i % 2 == 0:
        sum = sum + i

print("The sum is {0}".format(sum))

File: Code/R/problem002/euler002.R
#!/usr/bin/env Rscript

fibo <- function(n) if (n == 0 || n == 1) 1 else (fibo(n-1) + fibo(n - 2))

answer <- 0

counter <- 2
fib <- fibo(counter)
answer <- answer + fib
while ( fib < 4000000 ) {
  counter <- counter + 1
  fib <- fibo(counter)
  if (fib %% 2 == 0) answer <- answer + fib
}

print(answer)

# > source("./euler002.R")
# > fibo(0)
# [1] 1
# > fibo(1)
# [1] 1
# > fibo(2)
# [1] 2
# > fibo(3)
# [1] 3
# > fibo(4)
# [1] 5
# > fibo(5)
# [1] 8
# > fibo(10)
# [1] 89
# > fibo(9)
# [1] 55
# > fibos <-c(fibo(2))
# > typeof(fibos)
# [1] "double"
# > class(fibos)
# [1] "numeric"
# > ?append
# > append(fibos, fibo(5))
# [1] 2 8

